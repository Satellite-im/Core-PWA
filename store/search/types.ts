import { SearchResult } from '~/types/search/search'

export interface SearchState {
  query: string
  result: SearchResult | null
}
