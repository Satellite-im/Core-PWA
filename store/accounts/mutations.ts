import { AccountsState, RegistrationStatus } from './types'
import { User } from '~/libraries/Iridium/users/types'

const mutations = {
  setPin(state: AccountsState, pin: string) {
    state.pin = pin
  },
  setStorePin(state: AccountsState, storePin: boolean) {
    state.storePin = storePin
  },
  setRegistry(state: AccountsState, registry: boolean) {
    state.registry = registry
  },
  setPinHash(state: AccountsState, pinHash: string) {
    state.pinHash = pinHash
  },
  unlock(state: AccountsState, pin: string) {
    state.locked = false
    state.pin = pin
  },
  lock(state: AccountsState) {
    state.locked = true
  },
  setEncryptedPhrase(state: AccountsState, encryptedPhrase: string) {
    state.encryptedPhrase = encryptedPhrase
  },
  setPhrase(state: AccountsState, decryptedPhrase: string) {
    state.phrase = decryptedPhrase
  },
  setActiveAccount(state: AccountsState, activeAccountPubkey: string) {
    state.active = activeAccountPubkey
    state.initialized = true
  },
  setProfilePicture(state: AccountsState, image: string) {
    if (state.details) {
      state.details.photoHash = image
    }
  },
  setUserDetails(state: AccountsState, details: User) {
    state.details = {
      did: details.did,
      name: details.name,
      status: details.status,
      photoHash: details.photoHash,
    }
    state.welcomePopup = true
  },
  dismissWelcome(state: AccountsState) {
    state.welcomePopup = false
  },
  setUserPeerId(state: AccountsState, peerId: string) {
    if (state.details) {
      state.details = {
        ...state.details,
        peerId,
      }
    }
  },
  setRegistrationStatus(
    state: AccountsState,
    registrationStates: RegistrationStatus,
  ) {
    state.registrationStatus = registrationStates
  },
  setLastVisited(state: AccountsState, lastVisited: string) {
    state.lastVisited = lastVisited
  },
  setAdapter(state: AccountsState, adapter: string) {
    state.adapter = adapter
  },
  setEntropy(state: AccountsState, entropyMessage: string) {
    state.entropyMessage = entropyMessage
  },
}

export default mutations
