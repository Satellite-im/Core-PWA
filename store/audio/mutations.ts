import { AudioState } from './types'

const mutations = {
  mute(state: AudioState) {
    state.muted = !state.muted
  },
  toggleMute(state: AudioState) {
    state.muted = !state.muted
  },
  deafen(state: AudioState) {
    const isDeafened = state.deafened
    state.previousVolume = isDeafened ? state.previousVolume : state.volume
    state.volume = isDeafened ? state.previousVolume : 0
    state.deafened = !isDeafened
  },
  setVolume(state: AudioState, volume: Number) {
    state.previousVolume = volume
    state.volume = volume
  },
  setSystemVolume(state: AudioState, systemVolume: Number) {
    state.sounds.system = systemVolume
  },
  setInputVolume(state: AudioState, inputVolume: Number) {
    state.inputVolume = inputVolume
  },
  setSoundLevel(
    state: AudioState,
    sound: {
      inboundMedia: Number
      outboundMedia: Number
      system: Number
    },
  ) {
    state.sounds = sound
  },
}

export default mutations
