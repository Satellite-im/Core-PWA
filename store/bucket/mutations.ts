import { BucketState } from './types'
import { FilSystem } from '~/libraries/Files/FilSystem'

const mutations = {
  bucketInitialized(state: BucketState, status: boolean) {
    state.initialized = status
  },
  updateFileSystem(state: BucketState, fs: FilSystem) {
    state.fileSystem = fs
  },
  openDirectory(state: BucketState, name: string) {
    state.fileSystem.openDirectory(name)
  },
  createDirectory(state: BucketState, name: string) {
    state.fileSystem.createDirectory(name)
  },
  goBackToDirectory(state: BucketState, name: string) {
    state.fileSystem.goBackToDirectory(name)
  },
  createFile(state: BucketState, name: string) {
    state.fileSystem.createFile(name)
  },
}

export default mutations
