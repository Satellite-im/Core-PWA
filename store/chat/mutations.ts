import { ChatState } from './types'
import { ReplyObj, ChatTextObj } from '~/types/chat/chat'
import { UploadDropItemType } from '~/types/files/file'

const mutations = {
  chatText(state: ChatState, req: ChatTextObj) {
    state.chatTexts = state.chatTexts.some((item) => item.userId === req.userId)
      ? state.chatTexts.map((item) => {
          if (item.userId === req.userId) {
            return { ...item, value: req.value }
          }
          return item
        })
      : state.chatTexts.concat(req)
  },
  setChatReply(state: ChatState, req: ReplyObj) {
    state.replies = state.replies.some((item) => item.replyId === req.replyId)
      ? state.replies.map((item) => {
          if (item.replyId === req.replyId) {
            return { ...item, value: req.value }
          }
          return item
        })
      : state.replies.concat(req)
  },
  addFile(
    state: ChatState,
    {
      file,
      address,
    }: {
      file: UploadDropItemType
      address: string
    },
  ) {
    state.files[address]
      ? state.files[address].push(file)
      : (state.files[address] = [file])
  },
  setFiles(
    state: ChatState,
    {
      files,
      address,
    }: {
      files: UploadDropItemType[]
      address: string
    },
  ) {
    state.files[address] = files
  },
  deleteFiles(state: ChatState, address: string) {
    delete state.files[address]
  },
  setCountError(state: ChatState, countError: Boolean) {
    state.countError = countError
  },
  setAlertNsfw(state: ChatState, alertNsfw: Boolean) {
    state.alertNsfw = alertNsfw
  },
  setShowFilePreview(state: ChatState, showFilePreview: Boolean) {
    state.showFilePreview = showFilePreview
  },
}

export default mutations
