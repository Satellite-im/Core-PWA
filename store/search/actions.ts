import { Commit } from 'vuex'
import { DataStateType } from '../dataState/state'
import { searchResult } from '~/mock/search'

interface SearchArgument {
  commit: Commit
  state: any
}
export default {
  handler: () => {},
  async search({ commit, state }: SearchArgument) {
    if (
      state.dataState.files === DataStateType.Loading ||
      state.dataState.files === DataStateType.Updating
    ) {
      return
    }
    if (state.dataState.files === DataStateType.Empty) {
      commit('setDataState', { key: 'search', value: DataStateType.Loading })
    } else {
      commit('setDataState', { key: 'search', value: DataStateType.Updating })
    }
    await new Promise((resolve) => setTimeout(resolve, 3000))
    commit('search', searchResult)
    commit('setDataState', { key: 'search', value: DataStateType.Ready })
  },
}
