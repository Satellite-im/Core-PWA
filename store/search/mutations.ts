import { SearchState } from './types'
import { SearchResult } from '~/types/search/search'

const mutations = {
  search(state: SearchState, result: SearchResult) {
    state.result = result
  },
  setSearchQuery(state: SearchState, query: string) {
    state.query = query
  },
}

export default mutations
