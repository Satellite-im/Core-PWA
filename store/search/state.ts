import { SearchResult } from '~/types/search/search'

interface SearchState {
  query: string
  result: SearchResult | null
}

const InitalSearchState = (): SearchState => ({
  query: '',
  result: null,
})

export default InitalSearchState
