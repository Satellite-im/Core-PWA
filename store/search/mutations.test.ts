import { expect } from '@jest/globals'
import * as mutations from '~/store/search/mutations'

describe('mutations.default.setSearchQuery', () => {
  test('0', () => {
    const object2: any = [
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 100,
        type: 'number',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: 'Edmond',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Elio',
        value: 'elio@example.com',
      },
      {
        name: 'Jean-Philippe',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'elio@example.com',
        value: 'elio@example.com',
      },
      {
        name: 'Michael',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'elio@example.com',
        value: 'elio@example.com',
      },
      {
        name: 'George',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Elio',
        value: 'Elio',
      },
    ]
    const result: any = mutations.default.setSearchQuery(
      {
        query: 'DROP TABLE tmp;',
        result: {
          query:
            "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
          recommend: { key1: object, key3: [], key0: [], key2: [] },
          data: {
            pageInfo: { totalRows: 30, perPage: -1.0, pageIndex: -1 },
            list: object2,
          },
        },
      },
      'UPDATE Projects SET pname = %s WHERE pid = %s',
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.setSearchQuery(
      { query: 'UPDATE Projects SET pname = %s WHERE pid = %s', result: null },
      "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const object2: any = [
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: -100,
        type: 'number',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: 1,
        type: 'number',
        user: undefined,
      },
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 1,
        type: 'array',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Dillenberg',
        value: 'Dillenberg',
      },
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: undefined,
        key: 'elio@example.com',
        value: 'elio@example.com',
      },
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Dillenberg',
        value: 'elio@example.com',
      },
    ]
    const result: any = mutations.default.setSearchQuery(
      {
        query: 'UPDATE Projects SET pname = %s WHERE pid = %s',
        result: {
          query:
            "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
          recommend: { key1: object, key0: [], key2: [] },
          data: {
            pageInfo: { totalRows: 5, perPage: 0.5, pageIndex: 0 },
            list: object2,
          },
        },
      },
      "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const object2: any = [
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 1,
        type: 'array',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: 'Michael',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/ky_128.jpg',
        key: 'elio@example.com',
        value: 'Dillenberg',
      },
      {
        name: 'George',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/joannefournier_128.jpg',
        key: 'Dillenberg',
        value: 'Elio',
      },
      {
        name: 'Jean-Philippe',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Dillenberg',
        value: 'elio@example.com',
      },
      {
        name: 'George',
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Elio',
        value: 'Dillenberg',
      },
    ]
    const result: any = mutations.default.setSearchQuery(
      {
        query: 'DELETE FROM Projects WHERE pid = %s',
        result: {
          query: 'DROP TABLE tmp;',
          recommend: { key1: object, key3: [], key0: [], key2: [] },
          data: {
            pageInfo: { totalRows: 4, perPage: 10.0, pageIndex: -1 },
            list: object2,
          },
        },
      },
      'UPDATE Projects SET pname = %s WHERE pid = %s',
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const object: any = [
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: 0,
        type: 'string',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: -100,
        type: 'array',
        user: undefined,
      },
    ]
    const result: any = mutations.default.setSearchQuery(
      {
        query: 'UNLOCK TABLES;',
        result: {
          query: 'DELETE FROM Projects WHERE pid = %s',
          recommend: {},
          data: {
            pageInfo: { totalRows: 10, perPage: 10.23, pageIndex: 0 },
            list: object,
          },
        },
      },
      'UPDATE Projects SET pname = %s WHERE pid = %s',
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.setSearchQuery(
      {
        query: '',
        result: {
          query: '',
          recommend: { key1: [], key0: [], key2: [] },
          data: {
            pageInfo: { totalRows: NaN, perPage: NaN, pageIndex: NaN },
            list: [],
          },
        },
      },
      '',
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.search', () => {
  test('0', () => {
    const object2: any = [
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: -5.48,
        type: 'array',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: 100,
        type: 'object',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: -100,
        type: 'object',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 100,
        type: 'array',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/ky_128.jpg',
        key: 'Elio',
        value: 'Dillenberg',
      },
    ]
    const result: any = mutations.default.search(
      { query: 'UNLOCK TABLES;', result: null },
      {
        query:
          "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
        recommend: { key0: object },
        data: {
          pageInfo: { totalRows: 4, perPage: 0.5, pageIndex: -100 },
          list: object2,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const object2: any = [
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: -100,
        type: 'number',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: -100,
        type: 'object',
        user: undefined,
      },
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: -100,
        type: 'object',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 0,
        type: 'object',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: 'Anas',
        address: '192.168.1.5',
        avatar: undefined,
        key: 'elio@example.com',
        value: 'Elio',
      },
      {
        name: 'Michael',
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Dillenberg',
        value: 'Dillenberg',
      },
      {
        name: 'Pierre Edouard',
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Elio',
        value: 'elio@example.com',
      },
    ]
    const object4: any = [
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 100,
        type: 'object',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: 1,
        type: 'array',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 100,
        type: 'object',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 100,
        type: 'array',
        user: undefined,
      },
    ]
    const object3: any = [
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: undefined,
        key: 'Elio',
        value: 'elio@example.com',
      },
    ]
    const result: any = mutations.default.search(
      {
        query: 'DELETE FROM Projects WHERE pid = %s',
        result: {
          query: 'UNLOCK TABLES;',
          recommend: { key0: object, key2: [], key4: [], key3: [], key1: [] },
          data: {
            pageInfo: { totalRows: 5, perPage: 10.0, pageIndex: 0 },
            list: object2,
          },
        },
      },
      {
        query: 'UPDATE Projects SET pname = %s WHERE pid = %s',
        recommend: { key1: object3, key0: [], key2: [null] },
        data: {
          pageInfo: { totalRows: 10, perPage: -1.0, pageIndex: 100 },
          list: object4,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const object2: any = [
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: 100,
        type: 'array',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'elio@example.com',
        value: 'Dillenberg',
      },
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Dillenberg',
        value: 'Dillenberg',
      },
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Elio',
        value: 'elio@example.com',
      },
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Elio',
        value: 'elio@example.com',
      },
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'elio@example.com',
        value: 'Elio',
      },
    ]
    const result: any = mutations.default.search(
      { query: 'UPDATE Projects SET pname = %s WHERE pid = %s', result: null },
      {
        query: 'DROP TABLE tmp;',
        recommend: { key0: object, key2: [], key4: [], key3: [], key1: [] },
        data: {
          pageInfo: { totalRows: 10, perPage: -1.0, pageIndex: 100 },
          list: object2,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const object2: any = [
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 0,
        type: 'number',
        user: undefined,
      },
      {
        id: 'a85a8e6b-348b-4011-a1ec-1e78e9620782',
        at: -5.48,
        type: 'object',
        user: undefined,
      },
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 0,
        type: 'array',
        user: undefined,
      },
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: 100,
        type: 'array',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: 'Michael',
        address: '192.168.1.5',
        avatar: undefined,
        key: 'Elio',
        value: 'Elio',
      },
      {
        name: 'George',
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Elio',
        value: 'Dillenberg',
      },
      {
        name: 'Anas',
        address: '0.0.0.0',
        avatar: undefined,
        key: 'Elio',
        value: 'elio@example.com',
      },
    ]
    const object4: any = [
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 100,
        type: 'number',
        user: undefined,
      },
    ]
    const object3: any = [
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: 'https://cdn.fakercloud.com/avatars/zvchkelly_128.jpg',
        key: 'Elio',
        value: 'Elio',
      },
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Elio',
        value: 'Dillenberg',
      },
      {
        name: undefined,
        address: '192.168.1.5',
        avatar: 'https://cdn.fakercloud.com/avatars/joannefournier_128.jpg',
        key: 'Elio',
        value: 'Elio',
      },
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: 'https://cdn.fakercloud.com/avatars/joannefournier_128.jpg',
        key: 'elio@example.com',
        value: 'Elio',
      },
      {
        name: undefined,
        address: '0.0.0.0',
        avatar: 'https://cdn.fakercloud.com/avatars/ky_128.jpg',
        key: 'elio@example.com',
        value: 'Elio',
      },
    ]
    const result: any = mutations.default.search(
      {
        query: 'DELETE FROM Projects WHERE pid = %s',
        result: {
          query: 'UNLOCK TABLES;',
          recommend: { key0: object, key2: [], key4: [], key3: [], key1: [] },
          data: {
            pageInfo: { totalRows: 0, perPage: -0.5, pageIndex: -100 },
            list: object2,
          },
        },
      },
      {
        query: 'DROP TABLE tmp;',
        recommend: { key0: object3, key2: [], key4: [], key3: [], key1: [] },
        data: {
          pageInfo: { totalRows: 3.0, perPage: 1.0, pageIndex: 1 },
          list: object4,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const object2: any = [
      {
        id: '03ea49f8-1d96-4cd0-b279-0684e3eec3a9',
        at: -100,
        type: 'number',
        user: undefined,
      },
      {
        id: '7289708e-b17a-477c-8a77-9ab575c4b4d8',
        at: 100,
        type: 'object',
        user: undefined,
      },
    ]
    const object: any = [
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Dillenberg',
        value: 'Elio',
      },
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/joannefournier_128.jpg',
        key: 'elio@example.com',
        value: 'elio@example.com',
      },
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/akmalfikri_128.jpg',
        key: 'Elio',
        value: 'Elio',
      },
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Elio',
        value: 'Dillenberg',
      },
      {
        name: undefined,
        address: undefined,
        avatar: 'https://cdn.fakercloud.com/avatars/weavermedia_128.jpg',
        key: 'Elio',
        value: 'Elio',
      },
    ]
    const result: any = mutations.default.search(
      { query: 'DROP TABLE tmp;', result: null },
      {
        query:
          "SELECT * FROM Movies WHERE Title=’Jurassic Park’ AND Director='Steven Spielberg';",
        recommend: { key0: object },
        data: {
          pageInfo: { totalRows: 5, perPage: 0.0, pageIndex: 0 },
          list: object2,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.search(
      { query: '', result: null },
      {
        query: '',
        recommend: { key0: [] },
        data: {
          pageInfo: { totalRows: NaN, perPage: NaN, pageIndex: NaN },
          list: [],
        },
      },
    )
    expect(result).toMatchSnapshot()
  })
})
