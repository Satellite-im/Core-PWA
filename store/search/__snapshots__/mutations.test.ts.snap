// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.default.search 0 1`] = `undefined`;

exports[`mutations.default.search 1 1`] = `undefined`;

exports[`mutations.default.search 2 1`] = `undefined`;

exports[`mutations.default.search 3 1`] = `undefined`;

exports[`mutations.default.search 4 1`] = `undefined`;

exports[`mutations.default.search 5 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 0 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 1 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 2 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 3 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 4 1`] = `undefined`;

exports[`mutations.default.setSearchQuery 5 1`] = `undefined`;
